### PR TITLE
Porting CoreCLR JIT SelfHost tests

### PR DESCRIPTION
**Pull request by hoyMS**
Porting the following tests:

jit\methodical\arrays\misc\_dbgarrres.csproj
jit\methodical\arrays\misc\_relarrres.csproj
jit\methodical\arrays\misc\_speed_dbgarrres.csproj
jit\methodical\arrays\misc\_speed_relarrres.csproj
jit\methodical\explicit\basic\_dbgrefarg_c.csproj
jit\methodical\explicit\basic\_dbgrefarg_f4.csproj
jit\methodical\explicit\basic\_dbgrefarg_f8.csproj
jit\methodical\explicit\basic\_dbgrefarg_i1.csproj
jit\methodical\explicit\basic\_dbgrefarg_i2.csproj
jit\methodical\explicit\basic\_dbgrefarg_i4.csproj
jit\methodical\explicit\basic\_dbgrefarg_o.csproj
jit\methodical\explicit\basic\_dbgrefarg_s.csproj
jit\methodical\explicit\basic\_opt_dbgrefarg_c.csproj
jit\methodical\explicit\basic\_opt_dbgrefarg_f4.csproj
jit\methodical\explicit\basic\_opt_dbgrefarg_f8.csproj
jit\methodical\explicit\basic\_opt_dbgrefarg_i1.csproj
jit\methodical\explicit\basic\_opt_dbgrefarg_i2.csproj
jit\methodical\explicit\basic\_opt_dbgrefarg_i4.csproj
jit\methodical\explicit\basic\_opt_dbgrefarg_o.csproj
jit\methodical\explicit\basic\_opt_dbgrefarg_s.csproj
jit\methodical\explicit\basic\_opt_relrefarg_c.csproj
jit\methodical\explicit\basic\_opt_relrefarg_f4.csproj
jit\methodical\explicit\basic\_opt_relrefarg_f8.csproj
jit\methodical\explicit\basic\_opt_relrefarg_i1.csproj
jit\methodical\explicit\basic\_opt_relrefarg_i2.csproj
jit\methodical\explicit\basic\_opt_relrefarg_i4.csproj
jit\methodical\explicit\basic\_opt_relrefarg_o.csproj
jit\methodical\explicit\basic\_opt_relrefarg_s.csproj
jit\methodical\explicit\basic\_relrefarg_c.csproj
jit\methodical\explicit\basic\_relrefarg_f4.csproj
jit\methodical\explicit\basic\_relrefarg_f8.csproj
jit\methodical\explicit\basic\_relrefarg_i1.csproj
jit\methodical\explicit\basic\_relrefarg_i2.csproj
jit\methodical\explicit\basic\_relrefarg_i4.csproj
jit\methodical\explicit\basic\_relrefarg_o.csproj
jit\methodical\explicit\basic\_relrefarg_s.csproj

Modifying the following test to reflect correct configuration:

Jit\Methodical\arrays\dbg\gcarr.exe
Jit\Methodical\arrays\rel\gcarr.exe
Jit\Methodical\arrays\speed_dbg\gcarr.exe
Jit\Methodical\arrays\speed_rel\gcarr.exe

### CODE REVIEW

**Comment by hoyMS** [comment]
@pgavlin : please review this change. Thanks.

File: tests/src/JIT/Methodical/Arrays/misc/_relgcarr.csproj [hunk 1]
```diff
@@ -27,7 +27,7 @@
     </CodeAnalysisDependentAssemblyPaths>
   </ItemGroup>
   <PropertyGroup>
-    <DebugType></DebugType>
+    <DebugType>Full</DebugType>
```
Comment (pgavlin) [inline tests/src/JIT/Methodical/Arrays/misc/_relgcarr.csproj]: I don't think that this is correct: looking at the original build files, the rel build of this test should not have debug info enabled (i.e. `DebugType` should remain unset).

Comment (hoyMS) [inline tests/src/JIT/Methodical/Arrays/misc/_relgcarr.csproj]: Fixed.

**Comment by pgavlin** [comment]
LGTM aside from the comment re: `DebugType`.

**Pull request closed without merging**